src/__tests__/core/web_api/p2_api.test.js

```diff
@@ -159,4 +159,22 @@ describe('Auth0APIClient', () => {
     expect(mock.calls.length).toBe(1);
     expect(mock.calls[0]).toMatchSnapshot();
   });
+  describe('parseHash', () => {
+    it('should pass __enableImpersonation:false when options._enableImpersonation is not present', () => {
+      const client = getClient({});
+      client.parseHash('hash', 'cb');
+      const mock = getAuth0ClientMock();
+      const parseHashMock = mock.WebAuth.mock.instances[0].parseHash.mock;
+      expect(parseHashMock.calls.length).toBe(1);
+      expect(parseHashMock.calls[0]).toMatchSnapshot();
+    });
+    it('should pass __enableImpersonation when options._enableImpersonation===true', () => {
+      const client = getClient({ _enableImpersonation: true });
+      client.parseHash('hash', 'cb');
+      const mock = getAuth0ClientMock();
+      const parseHashMock = mock.WebAuth.mock.instances[0].parseHash.mock;
+      expect(parseHashMock.calls.length).toBe(1);
+      expect(parseHashMock.calls[0]).toMatchSnapshot();
+    });
+  });
 });
```

Comment:
Besides checking that the option is set on the auth0js client, you can also check that a real auth0js client can parse or throws an exception depending on the flag and on the `state` passed in the hash. (2 more tests)

Reply:
I'm not testing auth0.js internals inside lock